Surface_mesh/include/CGAL/Surface_mesh/Surface_mesh.h

```diff
@@ -2502,49 +2660,171 @@ class Property_map
   inline std::istream& sm_skip_comments( std::istream& in) {
       char c;
       in >> c;
-      if (c == '#')
-        in.ignore((std::numeric_limits<std::streamsize>::max)(), '\n');
-      else
-        in.putback(c);
+      while(c == '#')
+      {
+       in.ignore((std::numeric_limits<std::streamsize>::max)(), '\n');
+       in >> c;
+      }
+      in.putback(c);
       return in;
```

Comment:
I do not understand. That function is supposed to skip the comment until the end of the line. What does it do, now, with that modification?